controllers/machineset.go

```diff
@@ -98,179 +78,380 @@ func scaleMachineSet(aw *arbv1.AppWrapper, userRequestedInstanceType string, rep
+				totalReplicas := int32(replicas) + existingReplicas
+				copyOfaMachineSet.Spec.Replicas = &totalReplicas
+
+				// Check if MachineSet has replica label already and set isScaled = true
+				for k, v := range aMachineSet.Labels {
+					if k == fmt.Sprintf("instascale.codeflare.dev/%s", aw.Name) {
+						if v == strconv.FormatInt(int64(replicas), 10) {
+							isScaled = true
+						}
+					}
+				}
+
+				// Label Selector which will list machines that belong to the specific Machine Set
+				instanceLabelSelector := labels.SelectorFromSet(labels.Set(map[string]string{
+					"machine.openshift.io/instance-type":          userRequestedInstanceType,
+					"machine.openshift.io/cluster-api-machineset": aMachineSet.Name,
+				}))
+				instanceListOptions := &client.ListOptions{
+					LabelSelector: instanceLabelSelector,
+				}
+
+				// List all machines currently in the specific MachineSet
+				allExistingMachinesInMs := machinev1beta1.MachineList{}
+				if err := r.List(ctx, &allExistingMachinesInMs, instanceListOptions); err != nil {
+					klog.Infof("Error listing Machines: %s", err)
+				}
+
+				/*
+				   A for loop that will list machines with the AppWrapper label and apply the label
+				   if the count of machines with labels does not match the number of requested replicas
+				*/
+				machinesWithAwLabel := machinev1beta1.MachineList{}
+				if len(allExistingMachinesInMs.Items) > 0 {
+					for _, machine := range allExistingMachinesInMs.Items {
+						err := r.List(ctx, &machinesWithAwLabel, listOptions)
+						if err != nil {
+							klog.Infof("Error listing machines: %v", err)
+						}
+						machinePhase := machine.Status.Phase
+
+						if len(machinesWithAwLabel.Items) != replicas && (*machinePhase == "Running") {
+							nodeName := machine.Status.NodeRef.Name
+							r.addLabelToMachine(ctx, aw, machine.Name)
+							r.addLabelToNode(ctx, aw, nodeName)
+						}
+					}
+				}
+
+				// Get a list of running machines with the AppWrapper label
+				runningAwMachines = r.filterAwMachines(ctx, aw, userRequestedInstanceType)
+				canScaleMachineSets := r.canScaleMachineset(ctx, aw, demandMapPerInstanceType)
+				if (!isScaled || len(runningAwMachines) != replicas) && canScaleMachineSets {
+					// A loop to get the names of existing machines in the MachineSet
+					var existingMachinesOwned []string
+					for _, machine := range allExistingMachinesInMs.Items {
+						machinePhase := machine.Status.Phase
+						if !contains(existingMachinesOwned, machine.Name) && (*machinePhase != "Deleting") {
+							existingMachinesOwned = append(existingMachinesOwned, machine.Name)
+						}
+					}
+					klog.Infof("Existing machines owned in %s are %v", copyOfaMachineSet.Name, existingMachinesOwned)
+
+					// Apply the requested replica count label to the MachineSet
+					copyOfaMachineSet.Labels[fmt.Sprintf("instascale.codeflare.dev/%s", aw.Name)] = strconv.FormatInt(int64(replicas), 10)
```

Comment:
What about two AppWrappers with the same name, in different namespaces?

Reply:
These labels are now set to `"instascale.codeflare.dev/<appwrapper_name>-<appwrapper_namespace>=replicas` so they are more unique.